### PR TITLE
Enhancement/942 auth proxy setup

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #942

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by tofumatt — APPROVED** [review]
Looks good to me! 👍🏻

File: includes/Core/Authentication/Google_Proxy.php [hunk 1]
```diff
@@ -54,6 +54,107 @@ public function __construct( Context $context ) {
...
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Array of supported features.
+	 */
+	private function get_supports() {
+		$supports = array(
+			'credentials_retrieval',
+			'short_verification_token',
+			'user_input_flow',
```
Comment (aaemnnosttv) [inline includes/Core/Authentication/Google_Proxy.php:79]: We need to update this to also include the feature which was just added here:
https://github.com/google/site-kit-wp/blob/baec79d391492896a1ac4d0b41a6c93ac67ccf5c/includes/Core/Authentication/Clients/OAuth_Client.php#L921-L924

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @eugene-manuilov – this looks great overall, there's just one thing which is missing (see below) as well as a merge conflict.

Once addressed, please double check that this works with the new user input feature.

**Comment by eugene-manuilov** [comment]
@aaemnnosttv should be good now.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks!

**Pull request merged**